### ISSUE TITLE
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module

### ISSUE THREAD

**Issue by 2metres** [FormidableLabs/victory#1957]
# Bugs and Questions

### Checklist

- [x] This is not a `victory-native` specific issue. (Issues that only appear in `victory-native` should be opened [here](https://github.com/FormidableLabs/victory-native/issues/new))
- [x] I have read through the [FAQ](https://formidable.com/open-source/victory/docs/faq) and [Guides](https://formidable.com/open-source/victory/guides/) before asking a question
- [x] I am using the latest version of Victory
- [x] I've searched open issues to make sure I'm not opening a duplicate issue

### Enviroment
node v14.17.1
yarn 3.0.1
next.js 11.1.0
webpack 5

### The Problem

Since upgrading to v35.11.3 I am intermittently seeing this error

```sh
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /node_modules/d3-interpolate/src/index.js
require() of ES modules is not supported.
require() of /node_modules/d3-interpolate/src/index.js from /node_modules/victory-core/lib/victory-animation/util.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /node_modules/d3-interpolate/package.json.

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1102:13)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:14)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:92:18)
    at Object.<anonymous> (/node_modules/victory-core/lib/victory-animation/util.js:12:22)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:14) {
  code: 'ERR_REQUIRE_ESM'
}
```

### Reproduction

Attempted to reproduce but cannot seem to find a root cause other than page reload

**Comment by ryan-roemer** [comment]
Ah, on a quick glance it looks like https://unpkg.com/browse/d3-interpolate@3.0.1/package.json has:

```js
"exports": {
  "umd": "./dist/d3-interpolate.min.js",
  "default": "./src/index.js"
}
```

Which means that in Node.js it's _going_ to use ESM irrespective of your Node mode (which is causing this error since it's in CJS mode).

What I think this means is:

- `d3-interpolate` no longer supports CJS in Node.js for any SSR code.
- We probably can convert our code from a static `import` in source (which becomes a `require` in CJS output like here https://unpkg.com/browse/victory-core@35.11.3/lib/victory-animation/util.js - `var _d3Interpolate = require("d3-interpolate");`)

The awkward part IIRC the "compatible" thing in ESM to do is switch to a lazy dynamic import `import()` and without top-level `await` (I forget if we can guarantee to have this in early Node versions) it would be potentially something like:

```js
const getInterpolate = async () => {
  // TODO: Maybe memoize the import()
  const { interpolate } = await import("d3-interpolate");
  return interpolate; 
};
```

Which unfortunately to just _get_ `interpolate` you now need to be async 😬 .  There might be alternative solutions, but that's a first stab...

**Comment by achudars** [comment]
We started seeing this issue today with 35.9.x, 35.11.x

![image](https://user-images.githubusercontent.com/1721454/131702964-490a40ce-42bd-4c21-8855-8514f592acd1.png)

We're using victory, React and jest

We tried using `transformIgnorePatterns` but that didn't help

![image](https://user-images.githubusercontent.com/1721454/131703325-99e5e2d4-e575-47aa-aaaf-ce19f78b16a1.png)

The only thing that seemed to work but was tedious, was to override transitive dependencies.

So we had to override each transitive dependency that `victory` uses one by one: victory-core, victory-charts, victory-....

Hoping for the fix soon

**Comment by arbejd** [comment]
In the meantime, does anyone have a workaround centered around next-transpile-modules ? Im working on it but seems like nothing is working for me right now ^^

**Comment by arbejd** [comment]
For future reference: I downgraded victory all the way to 34.3.12 to get it to work again. It all relies on victory-core that has ^35.x.x in their respective packages, meaning this issue comes in for all versions >34.x.x

So its either that or @achudars overriding to get it to work today.

**Comment by boygirl** [comment]
I have a bandaid fix here that I will release today: https://github.com/FormidableLabs/victory/pull/1958

Dealing with async functions within our animation functions just to import a small interpolator utility is rough! I'm going to look into alternatives.

**Comment by boygirl** [comment]
Resolved in `victory@35.11.4`

**Comment by achudars** [comment]
Can confirm, version `victory@35.11.4` fixed it for us too. Thank you for the speedy resolution @boygirl !
